File: benchmarks/math/prime_count.js
// Prime checking - JavaScript
// Tests conditional logic and mathematical operations

function isPrime(n) {
    if (n < 2) {
        return false;
    }
    if (n === 2) {
        return true;
    }
    if (n % 2 === 0) {
        return false;
    }
    
    for (let i = 3; i * i <= n; i += 2) {
        if (n % i === 0) {
            return false;
        }
    }
    return true;
}

// Count primes up to 10000
let count = 0;
for (let i = 2; i <= 10000; i++) {
    if (isPrime(i)) {
        count++;
    }
}

console.log(count);

File: benchmarks/math/prime_count.py
# Prime checking - Python
# Tests conditional logic and mathematical operations

def is_prime(n):
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    
    i = 3
    while i * i <= n:
        if n % i == 0:
            return False
        i += 2
    return True

# Count primes up to 10000
count = 0
for i in range(2, 10001):
    if is_prime(i):
        count += 1

print(count)